internal/peermanager/peerwantmanager.go

```diff
@@ -60,6 +64,14 @@ func (pwm *peerWantManager) removePeer(p peer.ID) {
 		pwm.wantBlockGauge.Dec()
 	}
 
+	// Clean up the reverse index
+	for _, c := range pws.wantHaves.Keys() {
```

Comment:
Looking at this again, we should actually be using `pws.wantHaves.ForEach(...)` to avoid copying the entire set.